### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by V-Blaze**
These are the changes in this pull request:

-Implemented the composition design pattern.
-Created Nameable, and Decorate class.
-Created TrimmerDecorator, and CapitalizeDecorator class and their methods.

### CODE REVIEW

File: captalize_decorate.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorate'
```
Comment (TedLivist) [inline captalize_decorate.rb]: - [ ] Kindly ensure you test your code. This import is incorrect. There is not file named `decorate`. What you have is `decorator`. Kindly fix that.

File: captalize_decorate.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+class CapitalizeDecorator < Decorate
+  def correct_name
+    decorate.correct_name.captalize
```
Comment (TedLivist) [inline captalize_decorate.rb]: - [ ] There is no variable named `decorate`. What you should have is `@nameable` as inherited from the Decorator class. Kindly fix that.

- [ ] You also spelt "capitalize" wrong.

  ```
  @nameable.correct_name.capitalize
  ```

File: trimmer_decorate.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require './decorate'
```
Comment (TedLivist) [inline trimmer_decorate.rb]: - [ ] Please make sure you test your code. This import is incorrect. There is no file named `decorate`. What you have is `decorator`. Kindly fix that.

File: trimmer_decorate.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+class TrimmerDecorator < Decorate
+  def correct_name
+    decorate.correct_name.lenght <= 10 ? decorate.correct_name : decorate.correct_name[0..9]
```
Comment (TedLivist) [inline trimmer_decorate.rb]: - [ ] There is no variable named `decorate`. You spelt `length` wrong also. Kindly fix that.

  ```
  @nameable.correct_name.length <= 10 ? @nameable.correct_name : @nameable.correct_name[0..9]
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @V-Blaze,

Good job so far!
✔️ Implementation of decorator class
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @V-Blaze , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted


## Highlights
No linter errors :heavy_check_mark: 
Great effort so far :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**